beacon_node/store/src/hot_cold_store.rs

```diff
@@ -2389,6 +2421,33 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
             })
     }
 
+    /// Fetch custody info from the cache.
+    /// If custody info doesn't exist in the cache,
+    /// try to fetch from the DB and prime the cache.
+    pub fn get_data_column_custody_info(&self) -> Result<Option<DataColumnCustodyInfo>, Error> {
+        let Some(data_column_custody_info) = self.block_cache.lock().get_data_column_custody_info()
+        else {
+            let bytes_opt = self.blobs_db.get_bytes(
+                DBColumn::BeaconDataColumnCustodyInfo,
+                DATA_COLUMN_CUSTODY_INFO_KEY.as_slice(),
+            )?;
+
+            let Some(bytes) = bytes_opt else {
+                return Ok(None);
+            };
+
+            let data_column_custody_info = Some(DataColumnCustodyInfo::from_ssz_bytes(&bytes)?);
```

Comment:
I think you might be able to use `StoreItem` here to factor out some of this stuff, but it isn't essential